torch/lib/libshm/manager.cpp

```diff
@@ -83,16 +83,19 @@ int main(int argc, char *argv[]) {
   setsid();  // Daemonize the process
 
   std::unique_ptr<ManagerServerSocket> srv_socket;
+  char manager_socket_filename[] = "/tmp/fileXXXXXX";
```

Comment:
isn't this dependent on some env variables, like $TMP, and not always `/tmp`?
also if we have the opportunity, `fileXXXXXX` should be `torch-shm-fileXXXXXX`

Reply:
Oh hm, I guess I can `getenv` it

Reply:
ideally, mimic whatever they do inside `std::filesystem::temp_directory_path`

Reply:
https://github.com/llvm-mirror/llvm/blob/0253f7506fa808312119393a633d03cde96cbc8f/lib/Support/Unix/Path.inc#L998-L1037

looks like it checks `"TMPDIR", "TMP", "TEMP", "TEMPDIR"` in that order